### PR TITLE
added wgsl_analyzer server

### PR DESCRIPTION
**Pull request by skwee357**
WGSL is a WebGPU Shader Language. It has an LSP here https://github.com/wgsl-analyzer/wgsl-analyzer/ and an example to run it from NVIM is here https://github.com/wgsl-analyzer/wgsl-analyzer/issues/3

### CODE REVIEW

File: lua/nvim-lsp-installer/servers/wgsl_analyzer/init.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+if not configs.wgsl_analyzer then
+    configs.wgsl_analyzer = {
+        default_config = {
+            cmd = { "wgsl_analyzer" },
+            filetypes = { "wgsl" },
+            root_dir = lspconfig.util.root_pattern(".git", "wgsl"),
+            settings = {}
+        }
+    }
+end
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/wgsl_analyzer/init.lua:15]: Would you mind opening a PR to lspconfig for adding this server configuration there 🙏? We don't manage any in this plugin

**Review by williamboman — COMMENTED** [review]
Awesome! There's a script `scripts/autogen-metadata.sh` that needs to be run (requires a POSIX environment). I could do this for you if it's giving you issues

File: lua/nvim-lsp-installer/servers/wgsl_analyzer/init.lua [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+local cargo = require "nvim-lsp-installer.core.managers.cargo"
+local lspconfig = require "lspconfig"
+local configs = require "lspconfig.configs"
+
+if not configs.wgsl_analyzer then
+    configs.wgsl_analyzer = {
+        default_config = {
+            cmd = { "wgsl_analyzer" },
+            filetypes = { "wgsl" },
+            root_dir = lspconfig.util.root_pattern(".git", "wgsl"),
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/wgsl_analyzer/init.lua:11]: I'm not familiar with WGSL, but this root dir looks a bit weird to me - is there commonly a file called `wgsl` in the root of WGSL projects?

**Comment by skwee357** [comment]
Hey @williamboman Thanks for the comments!
I've opened a PR for `lspconfig` 👉 https://github.com/neovim/nvim-lspconfig/pull/1925
I've also tried to run the script `scripts/autogen-metadata.sh` but I'm getting errors:

```
E5113: Error while calling lua chunk: scripts/autogen_metadata.lua:165: scripts/autogen_metadata.lua:49: module 'lspconfig.server_configurations.rescriptls' not found:
```

I couldn't find any documentation where I can troubleshoot that. Appreciate some help there!

**Comment by williamboman** [comment]
> I couldn't find any documentation where I can troubleshoot that. Appreciate some help there!

Hm, weird. The script relies on you having lspconfig installed in the default location (`$XDG_DATA_HOME`) via packer, maybe that's why. I pushed some commits!

**Review by williamboman — APPROVED** [review]
LGTM! Now we just need to wait for the lspconfig PR to land. (stylua is failing on `main` branch too so dw)

**Comment by skwee357** [comment]
Hey @williamboman ! The lspconfig PR was merged! 🎉
Let me know what can I do to help with this PR!

**Pull request merged**